lib/exlasticsearch/repo.ex

```diff
@@ -258,16 +291,79 @@ defmodule ExlasticSearch.Repo do
     response
   end
 
-  defp bulk_operation({:delete, %{__struct__: model} = struct}),
-    do: [%{delete: %{_id: Indexable.id(struct), _index: model.__es_index__(:index), _type: model.__doc_type__()}}]
-  defp bulk_operation({op_type, %{__struct__: model} = struct}) do
+  defp bulk_operation({:delete, _struct, _index} = instruction),
```

Comment:
This has become sufficiently complex to probably deserve a dedicated module (readability and all that).